flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/operations/PlannerQueryOperation.java

```diff
@@ -72,6 +75,13 @@ public String asSummaryString() {
                 "PlannerNode", Collections.emptyMap(), getChildren(), Operation::asSummaryString);
     }
 
+    @Override
+    public String asSerializableString() {
+        final RelToSqlConverter relToSqlConverter = new RelToSqlConverter(AnsiSqlDialect.DEFAULT);
```

Comment:
In any case we need try-catch around it. Or will e.g. Table API queries that take the traditional route properly serialize?

Reply:
We should not expose Calcite exceptions in this case.

Reply:
> Since PlannerQueryOperation comes right out of the planner we can also pass some context into it. Don't we have some RelToSqlConverter or Flink dialect somewhere?

I did not look thoroughly enough. Actually we do have https://github.com/apache/flink/blob/78aa02f106cfe261200d4efce3c09d2a2882ebec/flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/operations/converters/SqlNodeConverter.java#L91

I'll use this to create the SQL string in `PlannerQueryOperation`.